src/BotGO/RobotPlayer.java

```diff
@@ -69,20 +68,46 @@ static void runHQ() throws GameActionException {
             tryBuild(RobotType.MINER, dir);
     }
 
-    static void runMiner() throws GameActionException {
+    static void runMiner() throws GameActionException
+    {
+        if (hqLoc == null)
+        {
+           // Senses near by Robots
+           RobotInfo[] robots = rc.senseNearbyRobots();
+            for (RobotInfo robot : robots)
+            {
+                // Updates the location of HQ if one robots knows it
+                if(robot.type == RobotType.HQ && robot.team == rc.getTeam())
+                {
+                    hqLoc = robot.location;
+                }
+            }
+        }
+
         tryBlockchain();
-        tryMove(randomDirection());
-        if (tryMove(randomDirection()))
-            System.out.println("I moved!");
         // tryBuild(randomSpawnedByMiner(), randomDirection());
-        for (Direction dir : directions)
-            tryBuild(RobotType.FULFILLMENT_CENTER, dir);
+        //for (Direction dir : directions)
+        //    tryBuild(RobotType.FULFILLMENT_CENTER, dir);
```

Comment:
leaving inactive code in comments can be confusing to people reading code.

Reply:
I kept it in comments since I thought I might use it again.